commands/role-request.js

```diff
@@ -0,0 +1,39 @@
+const {
+  ActionRowBuilder,
+  ButtonBuilder,
+  ButtonStyle,
+  SlashCommandBuilder,
+} = require("discord.js");
+
+module.exports = {
+  data: new SlashCommandBuilder()
+    .setName("request-role")
+    .setDescription("Request a role."),
+  async execute(interaction) {
+    const row = new ActionRowBuilder().addComponents(
+      new ButtonBuilder()
+        .setCustomId("role-request--cyber-academy")
+        .setLabel("Cyber Academy")
+        .setStyle(ButtonStyle.Primary),
+      new ButtonBuilder()
+        .setCustomId("role-request--special-topics")
+        .setLabel("Special Topics")
+        .setStyle(ButtonStyle.Primary)
+    );
+
+    await interaction.reply({ content: "Choose a role: ", components: [row] });
+  },
+  async onMessageInteraction(interaction) {
+    if (interaction.customId === "role-request--cyber-academy") {
+      await interaction.reply({
```

Comment:
Looks good so far - now we need to actually give roles to people. You can hardcode them or preferably use config.json. You can use config.json like so:

```js
const config = require("../utilities/config");

const config_options = config["role-request"][1];

// In config.json:
{
  "role-request": {
    1: "roleIDhere",
  }
}
```